.travis.sh

```diff
@@ -60,19 +60,52 @@ tests_sequence() {
     )
 }
 
+tests_sequence_ios_sim() {
+    title "testing from workspace directory"
+    ( \
+        cd test-ws \
+        && cargo clean \
+        && $CARGO_DINGHY   -d $1 -p auto-ios-aarch64-sim test pass \
+        && ! $CARGO_DINGHY -d $1 -p auto-ios-aarch64-sim test fails \
+        && ! $CARGO_DINGHY -d $1 -p auto-ios-aarch64-sim test \
```

Comment:
I am new to this tool so I don't know what `-p` does or why it seems to be needed here. I think maybe we're working around a bug where cargo-dinghy isn't automatically choosing `-sim` when the host is aarch64, even though it does automatically choose the right thing for x86_64?

if this is indeed a bug, perhaps we should fix that bug so that we can DRY these two functions?

Reply:
I was just wondering about that. It's a bug. To verify locally, I ran `rustup target remove x86_64-appl-ios` (on my m1), then `cargo dinghy -d <sim id> test` and it failed because dinghy was compiling for `x86_64-apple-ios` rather than `aarch64-apple-ios-sim`

Reply:
> cargo dinghy -d <sim id> test and it failed because dinghy was compiling for x86_64-apple-ios rather than aarch64-apple-ios-sim

Maybe the best thing to do by default is query the target device and choose the architecture that matches the host, if available, and then choose the default target triple based on what's compatible for that device? If the target device is a real device then it would be aarch64-apple-ios and if it is a simulator then either aarch64-apple-ios-sim or x86_64-apple-ios? And do analogously if the target device is a tvos (simulator) or watchos (simulator).

Reply:
Also, I think the kind of improvement I'm suggesting in this thread could be postponed to a future PR. The duplication here is unfortunate but this is good progress that would unblock work on several issues.